spec/lib/miq_automation_engine/service_methods/miq_ae_service_generic_object_spec.rb

```diff
@@ -6,6 +6,56 @@ module MiqAeServiceGenericObjectSpec
     let(:service_service)  { MiqAeMethodService::MiqAeServiceService.find(service.id) }
     let(:service2)         { FactoryGirl.create(:service) }
     let(:service_service2) { MiqAeMethodService::MiqAeServiceService.find(service2.id) }
+    let(:method_name)      { 'fred' }
+    let(:definition) do
+      FactoryGirl.create(
+        :generic_object_definition,
+        :name       => "test_definition",
+        :properties => {
+          :methods => [method_name]
+        }
+      )
+    end
+
+    let(:go) do
+      FactoryGirl.create(
+        :generic_object,
+        :generic_object_definition => definition,
+        :name                      => 'test'
+      )
+    end
+
+    let(:user) { FactoryGirl.create(:user_with_group) }
+    let(:workspace) do
+      double('ws', :persist_state_hash => {},
+                   :ae_user            => user,
+                   :rbac_enabled?      => false,
+                   :root               => {})
+    end
+
+    let(:options) do
+      { :object_type   => "GenericObject",
+        :object_id     => go.id,
+        :instance_name => "GenericObject",
+        :user_id       => user.id,
+        :miq_group_id  => user.current_group.id,
+        :tenant_id     => user.current_tenant.id,
+        :attrs         => {:method_name => method_name}
+      }
+    end
+
+    describe "call method on generic object" do
+      before do
+        allow(MiqAeEngine::DrbRemoteInvoker).to receive('workspace').and_return(workspace)
+      end
+
+      it "calls into automate" do
+        svc_obj = MiqAeMethodService::MiqAeServiceGenericObject.find(go.id)
+        expect(MiqAeEngine).to receive('deliver').with(options).and_return(workspace)
```

Comment:
Not sure it makes sense to have `options` as a `let` variable here since it is not being used anywhere else.  Moving it here would make it easier to see what the test is checking for as well.